tests/functional/z_cluster/test_performance_profile_validation.py

```diff
@@ -36,28 +38,71 @@
 @pytest.mark.polarion_id("OCS-5656")
 @pytest.mark.polarion_id("OCS-5657")
 class TestProfileDefaultValuesCheck(ManageTest):
-    def test_validate_cluster_resource_profile(self):
+    @pytest.mark.parametrize(
+        argnames=["perf_profile"],
+        argvalues=[
+            pytest.param(*["performance"]),
+            pytest.param(*["lean"]),
+            pytest.param(*["balanced"]),
+        ],
+    )
+    @ignore_leftovers
+    def test_validate_cluster_resource_profile(self, perf_profile):
         """
         Testcase to validate osd, mgr, mon, mds and rgw pod memory and cpu values
         are matching with the predefined set of values post profile updation
 
         """
         pv_pod_obj = []
+        namespace = config.ENV_DATA["cluster_namespace"]
         log.info("Obtaining the performance profile values from the cluster")
         storage_cluster_name = config.ENV_DATA["storage_cluster_name"]
         storage_cluster = StorageCluster(
             resource_name=storage_cluster_name,
             namespace=config.ENV_DATA["cluster_namespace"],
         )
-        performance_profile = storage_cluster.data["spec"]["resourceProfile"]
-        if performance_profile == constants.PERFORMANCE_PROFILE_LEAN:
+        self.perf_profile = perf_profile
+
+        try:
+            exist_performance_profile = storage_cluster.data["spec"]["resourceProfile"]
+        except KeyError:
+            exist_performance_profile = None
+            pass
```

Comment:
Please add a log message or a comment about the reseason we passed the test here.